airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -119,6 +119,8 @@ def _apply_transformations(
                 return message_or_record_data
         elif isinstance(message_or_record_data, dict):
             record = message_or_record_data
+        elif isinstance(message_or_record_data, Record):
```

Comment:
Could you document the reasoning behind this change in the existing comment above? Definitely a confusing part of the CDK

Reply:
This will not be necessary anymore with https://github.com/airbytehq/airbyte/issues/27494 as the transformation will apply to the raw dict before we create the `Record` or `AirbyteMessage` object so no need to document for now. I'll ping you on the PR for https://github.com/airbytehq/airbyte/issues/27494 so that you can assess the new code

Reply:
Even better!